metashare/api/jobs.py

```diff
@@ -470,3 +477,74 @@ def populate_github_users(repository):
 
 
 populate_github_users_job = job(populate_github_users)
+
+
+def submit_review(*, user, task, data):
+    try:
+        review_sha = None
+        org = data["org"]
+        notes = data["notes"]
+        status = data["status"]
+        delete_org = data["delete_org"]
+
+        task.refresh_from_db()
+        if org:
+            review_sha = org.latest_commit
+        elif task.review_valid:
+            review_sha = task.review_sha
+
+        if not (task.pr_is_open and review_sha):
+            raise TaskReviewIntegrityError(_("Cannot submit review for this task."))
+
+        repo_id = task.project.repository.get_repo_id(user)
+        repository = get_repo_info(user, repo_id=repo_id)
+        pr = repository.pull_request(task.pr_number)
+
+        # The values in this dict are the valid values for the
+        # `state` arg to repository.create_status. We are not
+        # currently using all of them, because some of them make no
+        # sense for our system to add to GitHub.
+        state_for_status = {
+            # "": "pending",
+            # "": "error",
+            TASK_REVIEW_STATUS.Approved: "success",
+            TASK_REVIEW_STATUS["Changes requested"]: "failure",
+        }.get(status)
+
+        target_url = get_user_facing_url(
+            path=[
+                "repositories",
+                task.project.repository.slug,
+                task.project.slug,
+                task.slug,
+            ]
+        )
+
+        # We filter notes to string.printable to avoid problems
+        # GitHub has with emoji in status descriptions
+        printable = set(string.printable)
+        filtered_notes = "".join(filter(lambda c: c in printable, notes))
+        repository.create_status(
+            review_sha,
+            state_for_status,
+            target_url=target_url,
+            description=filtered_notes[:25],
```

Comment:
@wlonk I added this truncation since GH errors on long descriptions, and only displays a few chars anyway.